perf_bench_test.go

```diff
@@ -339,31 +428,51 @@ func getCPUIntenseRPS() int {
 	return result
 }
 
-func getWarmupTime() int {
+func getUnloadServiceTime() float64 {
 	var (
-		max          int
-		funcs        = strings.Split(*funcNames, ",")
-		serviceTimes = map[string]int{
+		sum         float64
+		funcs       = strings.Split(*funcNames, ",")
+		serviceTime = map[string]float64{
 			"helloworld":   1,
-			"chameleon":    20,
+			"chameleon":    23,
 			"pyaes":        1,
-			"image_rotate": 26,
-			"json_serdes":  16,
-			"lr_serving":   3,
+			"image_rotate": 0,
+			"json_serdes":  0,
+			"lr_serving":   0,
 			"cnn_serving":  70,
-			"rnn_serving":  26,
+			"rnn_serving":  0,
 		}
 	)
+
 	for _, funcName := range funcs {
-		delay := serviceTimes[funcName]
-		if max < delay {
-			max = delay
-		}
+		sum += serviceTime[funcName]
 	}
-	return max
+
+	return sum / float64(len(funcs))
 }
 
 func calculateRPS(vmNum int) int {
 	baseRPS := getCPUIntenseRPS()
 	return vmNum * baseRPS
 }
+
+func bindSocket(omitPid string) error {
```

Comment:
when the number of VMs < 32 (SMT cores, make it configurable), you can pin the first VM to core 0 (or 1) and always collect counters from that core. When VMs >=32, there is always going to be something running on that core so you should not any VM to particular cores (only to a socket). 

Note that core-socket affinity may be different on different machines. No need to account for it in the code but need to keep this in mind.

Reply:
Done